internal/blockchain/ethereum/ethereum.go

```diff
@@ -259,19 +260,22 @@ func (e *Ethereum) handleBatchPinEvent(ctx context.Context, msgJSON fftypes.JSON
 		Event: blockchain.Event{
 			Source:     e.Name(),
 			Name:       "BatchPin",
-			ProtocolID: sTransactionHash,
+			ProtocolID: fmt.Sprintf("%.12d/%.6d/%.6d", blockNumber, txIndex, logIndex),
 			Output:     dataJSON,
 			Info:       msgJSON,
 			Timestamp:  timestamp,
 		},
 	}
 
 	// If there's an error dispatching the event, we must return the error and shutdown
-	return e.callbacks.BatchPinComplete(batch, authorAddress)
+	return e.callbacks.BatchPinComplete(batch, sTransactionHash, authorAddress)
```

Comment:
Feels slightly odd that for custom contract events, we use a `blockchain.EventWithContext` that contains the `BlockchainTXID` inside it, but for this path, we use a `blockchain.Event` with the blockchain transaction hash passed beside it as a separate parameter. Maybe `BlockchainTXID` should just move into `blockchain.Event`?

Reply:
I could certainly do that

Reply:
I think that makes the most sense... Seems like we pull it out of every blockchain event.